base/bucket_gocb.go

```diff
@@ -1202,16 +1204,21 @@ func (bucket *CouchbaseBucketGoCB) UpdateXattr(k string, xattrKey string, exp ui
 		} else {
 			if cas == 0 {
 				// If the doc doesn't exist, set SubdocDocFlagMkDoc to allow us to write the xattr
-				mutateFlag = gocb.SubdocDocFlagMkDoc
+				if supportsTombstoneCreation {
+					mutateFlag = gocb.SubdocDocFlag(gocbcore.SubdocDocFlag(0x08)) | gocb.SubdocDocFlagAccessDeleted | gocb.SubdocDocFlagReplaceDoc
+				} else {
+					mutateFlag = gocb.SubdocDocFlagMkDoc
+				}
 			} else {
 				// Since the body _may_ not exist, we need to set SubdocDocFlagAccessDeleted
 				mutateFlag = gocb.SubdocDocFlagAccessDeleted
 			}
 		}
 
 		builder := bucket.Bucket.MutateInEx(k, mutateFlag, gocb.Cas(cas), exp).
-			UpsertEx(xattrKey, xv, gocb.SubdocFlagXattr).                                                // Update the xattr
+			UpsertEx(xattrKey, xv, gocb.SubdocFlagCreatePath|gocb.SubdocFlagXattr).                      // Update the xattr
 			UpsertEx(xattrCasProperty, "${Mutation.CAS}", gocb.SubdocFlagXattr|gocb.SubdocFlagUseMacros) // Stamp the cas on the xattr
+
```

Comment:
In the case where we don't support tombstone creation, we want to set the crc32c value based on the document body `{}`, instead of empty body (`DeleteCrc32c`).  That should just mean changing line 1224 to:
`if isDelete && supportsTombstoneCreation`